### PR TITLE
Add notifier provider page to docs v2

### PR DESCRIPTION
**Pull request by leehack**
Changes:

* Add (Async)Notifier Provider page
* Add Todos example with NotifierProvider.
* Add remote todos example with AsyncNotifierProvider
* Add `freezed` dependency in the website package
* Use `freezed` for Codegen examples.
* Some generated files are updated due to the dependency update.

### CODE REVIEW

File: website/docs/providers/notifier_provider.mdx [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+import { trimSnippet, AutoSnippet } from "../../src/components/CodeSnippet";
+
+[NotifierProvider] is a provider that is used to listen to and expose a
+[Notifier].  
+[NotifierProvider] along with [Notifier] is Riverpod's recommended solution
+for managing state which may change in reaction to a user interaction.
+
+It is typically used for:
+
+- exposing an **immutable** state which can change over time after reacting to
```
Comment (rrousselGit) [inline website/docs/providers/notifier_provider.mdx:19]: This isn't true anymore. Notifier supports both with `ref.notifyListeners()`

File: website/docs/providers/notifier_provider.mdx [hunk 2]
```diff
@@ -0,0 +1,36 @@
+---
+title: NotifierProvider
+---
+
+import Tabs from "@theme/Tabs";
```
Comment (rrousselGit) [inline website/docs/providers/notifier_provider.mdx:5]: This page probably should include AsyncNotifier too. They are the same, with AsyncNotifier being the most used one

File: website/docs/providers/notifier_provider.mdx [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+import TabItem from "@theme/TabItem";
+import CodeBlock from "@theme/CodeBlock";
+import todos from "./notifier_provider/todos";
+import todosConsumer from "!!raw-loader!/docs/providers/notifier_provider/todos/todos_consumer.dart";
+import { trimSnippet, AutoSnippet } from "../../src/components/CodeSnippet";
+
+[NotifierProvider] is a provider that is used to listen to and expose a
+[Notifier].  
+[NotifierProvider] along with [Notifier] is Riverpod's recommended solution
```
Comment (rrousselGit) [inline website/docs/providers/notifier_provider.mdx:14]: If we add such statement, it should be removed from StateNotifierProvider

File: website/docs/providers/notifier_provider/todos/index.tsx [hunk 4]
```diff
@@ -0,0 +1,9 @@
+import raw from "!!raw-loader!./raw.dart";
+import codegen from "!!raw-loader!./todos.dart";
```
Comment (rrousselGit) [inline website/docs/providers/notifier_provider/todos/index.tsx:2]: The file should be named `codegen.dart`

File: website/docs/providers/notifier_provider/todos/raw.dart [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+}
+
+// The Notifier class that will be passed to our NotifierProvider.
+// This class should not expose state outside of its "state" property, which means
+// no public getters/properties!
+// The public methods on this class will be what allow the UI to modify the state.
+class TodosNotifier extends Notifier<List<Todo>> {
+  // We initialize the list of todos to an empty list
+  @override
+  List<Todo> build() {
```
Comment (rrousselGit) [inline website/docs/providers/notifier_provider/todos/raw.dart:39]: I think it's a good opportunity to use AsyncNotifier, and have `build` do a network request

File: website/docs/providers/notifier_provider/todos/todos.dart [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+import 'package:riverpod_annotation/riverpod_annotation.dart';
+
+part 'todos.g.dart';
+
+/* SNIPPET START */
+
+// The state of our StateNotifier should be immutable.
+// We could also use packages like Freezed to help with the implementation.
+@immutable
+class Todo {
```
Comment (rrousselGit) [inline website/docs/providers/notifier_provider/todos/todos.dart:11]: Could you use Freezed for this class?

We're in the code-gen variant of the source, so using Freezed makes sense.

File: website/docs/providers/notifier_provider.mdx (hunk 1 above)
Comment (leehack) [reply website/docs/providers/notifier_provider.mdx]: removed **immutable**

File: website/docs/providers/notifier_provider.mdx (hunk 2 above)
Comment (leehack) [reply website/docs/providers/notifier_provider.mdx:5]: Updated the tittle as `(Async)NotifierProvider`

File: website/docs/providers/notifier_provider.mdx (hunk 3 above)
Comment (leehack) [reply website/docs/providers/notifier_provider.mdx]: Removed the statement in `StateNotifierProvider`

File: website/docs/providers/notifier_provider/todos/index.tsx (hunk 4 above)
Comment (leehack) [reply website/docs/providers/notifier_provider/todos/index.tsx]: done

File: website/docs/providers/notifier_provider/todos/raw.dart (hunk 5 above)
Comment (leehack) [reply website/docs/providers/notifier_provider/todos/raw.dart:39]: Haven't addressed it yet.
Do you think we should add a new example for AsyncNotifier on top of the Notifier example or just replace it?

File: website/docs/providers/notifier_provider/todos/todos.dart (hunk 6 above)
Comment (leehack) [reply website/docs/providers/notifier_provider/todos/todos.dart]: done

File: website/docs/providers/notifier_provider/todos/raw.dart (hunk 5 above)
Comment (leehack) [reply website/docs/providers/notifier_provider/todos/raw.dart:39]: Added an example with the AsyncNotifier.

**Comment by leehack** [comment]
@rrousselGit Addressed all review comments. Would you review another round?

**Comment by leehack** [comment]
@rrousselGit rebased for target branch update.

**Comment by rrousselGit** [comment]
LGTM

**Pull request merged**